src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -538,8 +540,119 @@ export class NotebookModel extends Disposable implements INotebookModel {
+				let showPreview = this.cells[index].showPreview;
+
+				//Get selection value from current cell
+				let newcellcontent = model.getValueInRange(selection);
+
+				//Get content after selection
+				let tailrange = range.setStartPosition(selection.endLineNumber, selection.endColumn);
+				let tailcellcontent = model.getValueInRange(tailrange);
+
+				//Get content before selection
+				let headrange = range.setEndPosition(selection.startLineNumber, selection.startColumn);
+				let headcontent = model.getValueInRange(headrange);
+
+				//Set content before selection if the selection is not the same as original content
+				if (headcontent.length) {
+					let headsource = source.slice(range.startLineNumber - 1, selection.startLineNumber - 1);
+					if (selection.startColumn > 1) {
+						partial = source.slice(selection.startLineNumber - 1, selection.startLineNumber)[0].slice(0, selection.startColumn - 1);
+						headsource = headsource.concat(partial.toString());
+					}
+					this.cells[index].source = headsource;
+				}
+
+				if (newcellcontent.length) {
+					newcell = this.createCell(cellType);
+					let newsource = source.slice(selection.startLineNumber - 1, selection.endLineNumber) as string[];
+					if (selection.startColumn > 1) {
+						partial = source.slice(selection.startLineNumber - 1)[0].slice(selection.startColumn - 1);
+						newsource.splice(0, 1, partial);
+					}
+					if (selection.endColumn !== source[selection.endLineNumber - 1].length) {
+						let splicestart = 0;
+						if (selection.startLineNumber === selection.endLineNumber) {
+							splicestart = selection.startColumn - 1;
+						}
+						let partial = source.slice(selection.endLineNumber - 1, selection.endLineNumber)[0].slice(splicestart, selection.endColumn - 1);
+						newsource.splice(newsource.length - 1, 1, partial);
+					}
+					newcell.source = newsource;
+					newcellindex++;
+					this.insertCell(newcell, newcellindex);
+				}
+
+				if (tailcellcontent.length) {
+					//tail cell will be of original cell type.
+					tailcell = this.createCell(this._cells[index].cellType);
+					let tailsource = source.slice(tailrange.startLineNumber - 1) as string[];
+					if (selection.endColumn > 1) {
+						partial = source.slice(tailrange.startLineNumber - 1, tailrange.startLineNumber)[0].slice(tailrange.startColumn - 1);
+						tailsource.splice(0, 1, partial);
+					}
+					tailcell.source = tailsource;
+					tailcellindex = newcellindex + 1;
+					this.insertCell(tailcell, tailcellindex);
+				}
+
+				//Delete the original cell if the selection begins at the start
+				if (!headcontent.length) {
+					this.deleteCell(this.cells[index]);
+				}
+				let activecell = newcell ? newcell : tailcell;
+				let activecellindex = newcell ? newcellindex : tailcellindex;
+
+				//make new cell Active
+				this.updateActiveCell(activecell);
+				this._contentChangedEmitter.fire({
+					changeType: NotebookChangeType.CellsModified,
+					cells: [activecell],
```

Comment:
Shouldn't this include all the cells that were added/updated?

Reply:
This is to make the cell active and out it in editable state so that users can continue editing them.